src/command_prechecks.ts

```diff
@@ -0,0 +1,49 @@
+import { areUserAndBotInSameVoiceChannel, getDebugLogHeader, sendErrorMessage } from "./helpers/discord_utils";
+import GameSession from "./structures/game_session";
+import MessageContext from "./structures/message_context";
+import { GameType, GuildTextableMessage } from "./types";
+import { IPCLogger } from "./logger";
+
+const logger = new IPCLogger("command_prechecks");
+
+export function inGameCommandPrecheck(message: GuildTextableMessage, gameSession: GameSession): boolean {
+    if (!gameSession) {
+        return false;
+    }
+
+    if (!areUserAndBotInSameVoiceChannel(message)) {
+        if (gameSession.gameType === GameType.ELIMINATION || gameSession.gameType === GameType.TEAMS) {
+            if (!gameSession.sessionInitialized) {
+                // The bot doesn't join the voice channel until after ,begin is called;
+                // players should still be able ,end before that happens in these game modes
+                return true;
+            }
+        }
+
+        logger.warn(`${getDebugLogHeader(message)} | User and bot are not in the same voice connection`);
+        sendErrorMessage(MessageContext.fromMessage(message), { title: "Wait...", description: "You must be in the same voice channel as the bot to use this command." });
+        return false;
+    }
+
+    return true;
+}
+
+export function debugServerPrecheck(message: GuildTextableMessage, _gameSession: GameSession): boolean {
+    const res = process.env.DEBUG_SERVER_ID === message.guildID;
```

Comment:
Nit: you can avoid the temporary variable here and in the function below

Reply:
Need the result of the variable to return the warning message if true

Reply:
Could return false if there's an error, and true if there isn't